### PR TITLE
assignVarFromValue - Add template expressions as value

### PR DESCRIPTION
**Pull request by thim81**
linked to #595 

Portman config:

```yaml
globals:
  stripResponseExamples: true
  variableCasing: snakeCase
assignVariables:
  - openApiOperation: GET::/crm/leads/{id}
    collectionVariables:
      - value: <tag>
        name: <tag>MonetaryAmount
```

Result in Postman in:
BEFORE

image

AFTER
image

### CODE REVIEW

File: src/application/variables/assignVarFromValue.ts [hunk 1]
```diff
@@ -44,7 +44,20 @@ export const assignVarFromValue = (
...
+      casing: globals?.variableCasing
+    }
+  })
+  const varValue =
+    varSetting?.value && hasTpl(varSetting.value)
+      ? generatedValue
+      : typeof varSetting.value === 'string'
+      ? `"${varSetting.value}"`
+      : varSetting.value
+  // const varValue = typeof varSetting.value === 'string' ? `"${varSetting.value}"` : varSetting.value
```
Comment (nicklloyd) [inline src/application/variables/assignVarFromValue.ts:60]: remove if not needed anymore...

Comment (thim81) [reply src/application/variables/assignVarFromValue.ts]: Cleaned-up and discovered that the result was not a "string", which was expected, which is now also corrected.

File: src/application/variables/assignVarFromValue.ts [hunk 2]
```diff
@@ -51,13 +51,8 @@ export const assignVarFromValue = (
...
   })
-  const varValue =
-    varSetting?.value && hasTpl(varSetting.value)
-      ? generatedValue
-      : typeof varSetting.value === 'string'
-      ? `"${varSetting.value}"`
-      : varSetting.value
-  // const varValue = typeof varSetting.value === 'string' ? `"${varSetting.value}"` : varSetting.value
+  const convVal = varSetting?.value && hasTpl(varSetting.value) ? generatedValue : varSetting.value
+  const varValue = typeof convVal === 'string' ? `"${convVal}"` : convVal
```
Comment (nicklloyd) [inline src/application/variables/assignVarFromValue.ts:55]: is a style thing, but just don't like not knowing what `convVal` refers too when you see it in isolation. We're not gaining by abbreviating variables name....

Comment (thim81) [reply src/application/variables/assignVarFromValue.ts:55]: Good comment, it was a temp variable, but even that should be descriptive enough.

Changed it to:
```
  const processedValue = varSetting?.value && hasTpl(varSetting.value) ? generatedValue : varSetting.value
  const varValue = typeof processedValue === 'string' ? `"${processedValue}"` : processedValue
```

Is that better?

**Review by nicklloyd — APPROVED** [review]

**Pull request merged**